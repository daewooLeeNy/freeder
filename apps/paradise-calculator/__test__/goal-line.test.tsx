import GoalLine from "@/app/goal-line/page";
import { render } from "@testing-library/react";

describe("낙원 기간 목표 화면", () => {
  it("기본값이 설정된 진입 페이지", () => {
    const { container } = render(<GoalLine />);
    expect(container).toMatchSnapshot("기본값");
  });
});
