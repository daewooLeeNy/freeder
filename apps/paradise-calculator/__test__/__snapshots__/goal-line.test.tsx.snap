// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`낙원 기간 목표 화면 기본값이 설정된 진입 페이지: 기본값 1`] = `
<div>
  <main
    class="w-full"
  >
    <h1
      class="text-lg font-bold"
    >
      🏖 낙원 시간(🏁) 계산기 
      <span
        class="text-sm font-medium"
      >
        (은퇴시기 계산하기)
      </span>
    </h1>
    <p
      class="text-sm"
    >
      여러분의 낙원(경제적 자유)까지 얼마나 걸리는지 계산해보세요. 여러분은 이미 낙원🏖에 한발자국👟👟 다가갔습니다.
    </p>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow my-4"
    >
      <div
        class="flex flex-col space-y-1.5 p-6 relative"
      >
        <div
          class="flex justify-between w-full"
        >
          <h3
            class="font-semibold leading-none tracking-tight flex-1"
          >
            목표 설정
          </h3>
          <div
            class="flex flex-1 items-center justify-end gap-2 xs:mr-8"
          >
            <label
              class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
              for="dividend-goal-main"
            >
              배당 목표
            </label>
            <button
              aria-checked="false"
              class="peer inline-flex h-5 w-9 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent shadow-sm transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
              data-state="unchecked"
              id="dividend-goal-main"
              role="switch"
              type="button"
              value="on"
            >
              <span
                class="pointer-events-none block h-4 w-4 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-4 data-[state=unchecked]:translate-x-0"
                data-state="unchecked"
              />
            </button>
          </div>
        </div>
        <svg
          aria-controls="radix-:r0:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="lucide lucide-settings2 absolute top-[18px] right-7 hidden xs:block"
          data-state="closed"
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          type="button"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 7h-9"
          />
          <path
            d="M14 17H5"
          />
          <circle
            cx="17"
            cy="17"
            r="3"
          />
          <circle
            cx="7"
            cy="7"
            r="3"
          />
        </svg>
      </div>
      <div
        class="p-6 pt-0"
      >
        <div
          class="grid grid-cols-2 space-4 gap-4"
        >
          <div>
            <label
              class="block text-sm font-medium mb-1"
              for="targetAmount"
            >
              낙원(목표) 금액₩
            </label>
            <input
              class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
              id="targetAmount"
              placeholder="0"
              type="text"
              value="1,000,000,000"
            />
          </div>
          <div>
            <label
              class="block text-sm font-medium mb-1"
              for="investmentReturn"
            >
              목표 투자 수익률 (%)
            </label>
            <input
              class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
              id="investmentReturn"
              placeholder="0"
              type="text"
              value="10"
            />
          </div>
          <div>
            <label
              class="block text-sm font-medium mb-1"
              for="currentAssets"
            >
              현재 자산 (원)
            </label>
            <input
              class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
              id="currentAssets"
              placeholder="0"
              type="text"
              value="100,000,000"
            />
          </div>
          <div>
            <label
              class="block text-sm font-medium mb-1"
              for="annualSavings"
            >
              연간 저축액 (원)
            </label>
            <input
              class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
              id="annualSavings"
              placeholder="0"
              type="text"
              value="20,000,000"
            />
          </div>
          <div>
            <label
              class="block text-sm font-medium mb-1"
              for="dividendYield"
            >
              목표 달성 후 예상 배당 수익률 (%)
            </label>
            <input
              class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
              id="dividendYield"
              placeholder="0"
              type="text"
              value="5"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow mb-4"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <h3
          class="font-semibold leading-none tracking-tight"
        >
          결과
        </h3>
      </div>
      <div
        class="p-6 pt-0"
      >
        <div
          class="space-y-2"
        >
          <p
            class="text-lg font-semibold"
          >
            🏖낙원🏖 
            금액
             달성까지 약 
            <span
              class="font-semibold underline decoration-primary"
            >
              19
            </span>
            년이 걸립니다.
          </p>
          <p>
            달성 금액:  
            <a
              class="cursor-pointer"
              data-state="closed"
            >
              <span
                class="font-semibold underline decoration-primary"
              >
                ₩1,857,991,759
              </span>
            </a>
             
             (물가상승률: ↑3%)
          </p>
          <p>
            달성 후 예상 배당금: 
            <a
              data-state="closed"
            >
              <span
                class="font-semibold underline decoration-primary cursor-pointer"
              >
                ₩7,741,632
                /월 
              </span>
            </a>
            , 
            <span
              class="font-semibold underline decoration-primary"
            >
              ₩92,899,588
              /연
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow mb-4"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <h3
          class="font-semibold leading-none tracking-tight"
        >
          연도별 자산 증가 그래프
        </h3>
      </div>
      <div
        class="p-6 pt-0"
      >
        <div
          class="recharts-responsive-container"
          style="width: 100%; height: 300px; min-width: 0;"
        />
      </div>
    </div>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <h3
          class="font-semibold leading-none tracking-tight flex justify-between"
        >
          <div>
            년도별 자산 변동 테이블
          </div>
          <svg
            class="lucide lucide-chevrons-up-down w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m7 15 5 5 5-5"
            />
            <path
              d="m7 9 5-5 5 5"
            />
          </svg>
        </h3>
      </div>
      <div
        class="p-6 pt-0"
      >
        <div
          data-state="closed"
        >
          <div
            data-state="closed"
            hidden=""
            id="radix-:r3:"
            style=""
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
